File: java/producer/src/main/java/br/com/dynamiclight/push/PushConfiguration.java
package br.com.dynamiclight.push;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties(prefix = "push")
public class PushConfiguration {
    /**
     * Kafka topic to send push messages
     */
    private String topic;

    public String getTopic() {
        return topic;
    }

    public void setTopic(String topic) {
        this.topic = topic;
    }
}

File: kotlin/producer/src/main/kotlin/br/com/dynamiclight/push/PushConfiguration.kt
package br.com.dynamiclight.push

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "push")
data class PushConfiguration (
    /**
     * Kafka topic to send push messages
     */
    var topic: String? = null
)

